### ISSUE TITLE
"Play ringtone" does not work

### ISSUE THREAD

**Issue by MarcinOrlowski** [kevalpatel2106/android-ringtone-picker#1]
"Play ringtone" in demo app does not seem to cause any effects and app stays quiet in both picker and after selection.

**Comment by kevalpatel2106** [comment]
@MarcinOrlowski Did you enabled **Play Ringtone** switch?

**Comment by MarcinOrlowski** [comment]
Yes, no difference it was on or off

**Comment by egoliveira** [comment]
I had the same issue. I believe you have to reset the media player regardless if you are playing a media file or not.

RingTonePlayer.java, lines 36-39:

    if (mMediaPlayer.isPlaying()) {
        mMediaPlayer.stop();
    }
    
    mMediaPlayer.reset();

**Comment by jishindev** [comment]
Crashed when clicking on a different item in the list.

As @egoliveira mentioned, you should be resetting the media player every time.

Stacktrace -  

```
 java.lang.IllegalStateException
                      at android.media.MediaPlayer._setDataSource(Native Method)
                      at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1128)
                      at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1113)
                      at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1010)
                      at android.media.MediaPlayer.setDataSource(MediaPlayer.java:968)
                      at com.kevalpatel.ringtonepicker.RingTonePlayer.playRingtone(RingTonePlayer.java:41)
                      at com.kevalpatel.ringtonepicker.RingtonePickerDialog$2.onClick(RingtonePickerDialog.java:157)
                      at android.support.v7.app.AlertController$AlertParams$3.onItemClick(AlertController.java:1045)
                      at android.widget.AdapterView.performItemClick(AdapterView.java:310)
                      at android.widget.AbsListView.performItemClick(AbsListView.java:1145)
                      at android.widget.AbsListView$PerformClick.run(AbsListView.java:3066)
                      at android.widget.AbsListView$3.run(AbsListView.java:3903)
                      at android.os.Handler.handleCallback(Handler.java:739)
                      at android.os.Handler.dispatchMessage(Handler.java:95)
                      at android.os.Looper.loop(Looper.java:148)
                      at android.app.ActivityThread.main(ActivityThread.java:5417)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by kevalpatel2106** [comment]
This issue is fixed. Fix will be available in next release.

**Comment by kevalpatel2106** [comment]
This issue is fixed in [v1.1](https://github.com/kevalpatel2106/android-ringtone-picker/releases/tag/v1.1).
